test/jetstream_test.go

```diff
@@ -4547,9 +4547,29 @@ func TestJetStreamStreamPurge(t *testing.T) {
 				t.Fatalf("Expected %d messages, got %d", 100, state.Msgs)
 			}
 			mset.Purge()
-			if state := mset.State(); state.Msgs != 0 {
+			state := mset.State()
+			if state.Msgs != 0 {
 				t.Fatalf("Expected %d messages, got %d", 0, state.Msgs)
 			}
+			// Make sure first timestamp are reset.
+			if !state.FirstTime.IsZero() {
+				t.Fatalf("Expected the state's first time to be zero after purge")
+			}
```

Comment:
need to also fix last ts

```
			// Make sure timestamps are reset.
			if !state.FirstTime.IsZero() {
				t.Fatalf("Expected the state's first time to be zero after purge")
			}
			if !state.LastTime.IsZero() {
				t.Fatalf("Expected the state's last time to be zero after purge")
			}
```

Reply:
I had it that way but if I reset it I can not tell when the last message was received so reverted.

Reply:
hmm, but if its purged its concceptually as if a last message was never received.

Reply:
Kindof, we still track next first seq etc and like I said as I was playing with it for myself it seemed like I really wanted to know when the last message had been received regardless of if it had expired or been purged.